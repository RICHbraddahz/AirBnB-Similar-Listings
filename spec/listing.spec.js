import React from 'react';
import { shallow, mount, render } from 'enzyme'; 
import { shallowToJson } from 'enzyme-to-json';

import Listing from '../client/src/listing';
import Details from '../client/src/details';

describe('Listing', () => {
  const id = 199;
  const url = "mockUrl";
  const thumbnailImage = "mockImageUrl";
  const additionalDetails = {
    "title":"nostrum laboriosam in rerum officiis",
    "type":"Shared Room",
    "numBeds":2,
    "price":810,
    "numRatings":965,
    "avgStars":0.5,
  };

  const shallowWrapper = shallow(
    <Listing
      id={id}
      url={url}
      thumbnailImage={thumbnailImage}
      additionalDetails={additionalDetails}
    />
  );

  test('renders correctly', () => {
    expect(shallowToJson(shallowWrapper)).toMatchSnapshot();
  });

  test('renders one <img> element', () => {
    expect(shallowWrapper.find('img').length).toBe(1);
  })

  test('renders one Details component', () => {
    expect(shallowWrapper.find(Details).length).toBe(1);
  })
})
