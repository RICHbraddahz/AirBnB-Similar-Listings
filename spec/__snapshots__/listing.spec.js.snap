// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing renders correctly 1`] = `
<div
  className="similar-listing"
>
  <img
    src="mockImageUrl"
  />
  <Details
    avgStars={0.5}
    numBeds={2}
    numRatings={965}
    price={810}
    title="nostrum laboriosam in rerum officiis"
    type="Shared Room"
  />
</div>
`;
